src/server/routes/editor.js

```diff
@@ -250,47 +251,92 @@ router.get('/:id', (req, res, next) => {
 	);
 });
 
-router.get('/:id/revisions', (req, res, next) => {
+async function getOrderedRevisionForEditorPage(from, size, req) {
+	const {Note, Revision} = req.app.locals.orm;
+
+	const revisions = await new Revision()
+		.query('where', 'author_id', '=', parseInt(req.params.id, 10))
+		.orderBy('created_at', 'DESC')
+		.fetchPage({
+			limit: size,
+			offset: from
+		});
+	const revisionsJSON = revisions.toJSON();
+
+	const formattedRevisions = await Promise.all(
+		revisionsJSON.map(async rev => {
+			const note = await new Note()
+				.query('where', 'revision_id', '=', parseInt(rev.id, 10))
+				.fetchAll({require: false});
+			const noteJSON = note.toJSON();
+			const noteContent = noteJSON.map(currentNote => currentNote.content);
+
+			const {id: revisionId, ...otherProps} = rev;
+			return {entities: [], noteContent, revisionId, ...otherProps};
+		}));
+
+
+	const orderedRevisions = await utilis.getAssociatedEntityRevisions(formattedRevisions, req.app.locals.orm);
+
+	return orderedRevisions;
+}
+
+router.get('/:id/revisions', async (req, res, next) => {
 	const {Editor, TitleUnlock} = req.app.locals.orm;
-	new Editor({id: parseInt(req.params.id, 10)})
-		.fetch({
-			require: true,
-			withRelated: {
-				revisions(query) {
-					query.orderBy('id');
-				}
-			}
-		})
-		.then((editor) => getEditorTitleJSON(editor.toJSON(), TitleUnlock))
-		.then((editorJSON) => {
-			const props = generateProps(req, res, {
-				editor: editorJSON,
-				tabActive: 1
-			});
-			const markup = ReactDOMServer.renderToString(
-				<Layout {...propHelpers.extractLayoutProps(props)}>
-					<EditorContainer
-						{...propHelpers.extractEditorProps(props)}
-					>
-						<RevisionsTab
-							editor={props.editor}
-						/>
-					</EditorContainer>
-				</Layout>
-			);
-			res.send(target({
-				markup,
-				page: 'revisions',
-				props: escapeProps(props),
-				script: '/js/editor/editor.js'
-			}));
-		})
-		.catch(Editor.NotFoundError, () => {
-			throw new error.NotFoundError('Editor not found', req);
-		})
-		.catch(next);
+
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+
+	const orderedRevisions = await getOrderedRevisionForEditorPage(from, size, req);
+
+	const editor = await new Editor({id: req.params.id}).fetch();
```

Comment:
The mechanism should catch if there is no existing Editor here.
Try visiting an non-existing user's revisions: http://localhost:9099/editor/12789345/revisions
That's what the
```
.catch(Editor.NotFoundError, () => {
	throw new error.NotFoundError('Editor not found', req);
})
```
 block was doing. You'll have to recreate that.

I think you should  also do that before calling `getOrderedRevisionForEditorPage`, or do the check (does the user exist?) in `getOrderedRevisionForEditorPage` itself.